src/Uno.UI.Composition/Composition/ImageFrameProvider.skia.cs

```diff
@@ -0,0 +1,145 @@
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using SkiaSharp;
+
+namespace Microsoft.UI.Composition;
+
+internal sealed class ImageFrameProvider : IDisposable
+{
+	private readonly SKImage[] _images;
+	private readonly SKCodecFrameInfo[]? _frameInfos;
+	private readonly Timer? _timer;
+	private readonly Stopwatch? _stopwatch;
+	private readonly long _totalDuration;
+	private readonly Action? _onFrameChanged;
+
+	private int _currentFrame;
+	private bool _disposed;
+
+	private ImageFrameProvider(SKImage[] images, SKCodecFrameInfo[]? frameInfos, long totalDuration, Action? onFrameChanged)
```

Comment:
You should keep a wake ref on the `onFrameChanged` and auto-stop if the callback is being collected.

Reply:
Yup. Indeed.